search-app/src/cmr/search/services/humanizers/humanizer_report_service.clj

```diff
@@ -95,3 +105,88 @@
                        "concat humanized rows:" t3))))]
       (info "Create report " t4)
       (str string-writer))))
+
+(defn safe-generate-humanizers-report-csv
+  "This convenience function wraps the report generator in a try/catch."
+  [context]
+  (try
+    (generate-humanizers-report-csv context)
+    (catch Exception e
+      (warn (.getMessage e) "Returning empty report."))))
+
+(def report-cache-key
+  "The key used to store the humanizer report cache in the system cache map."
+  :report-cache)
+
+(defn get-cache
+  "A utility function for returning the system report cache instance."
+  [context]
+  (get-in context [:system :caches report-cache-key]))
+
+(defn read-report
+  "A convenience function wrapping the cache object to retreive report
+  data."
+  [context]
+  (.get-value (get-cache context)
+              csv-report-cache-key))
+
+(defn write-report
+  "A convenience function wrapping the cache object to retreive report
+  data."
+  [context]
+  (.set-value (get-cache context)
+              csv-report-cache-key
+              (safe-generate-humanizers-report-csv context)))
+
+(defconfig report-cache-consistent-timeout-seconds
+  "The number of seconds between when the report cache should check with cubby
+  for data consistency."
+  {:default (* 60 60 6) ; every 6 hours
```

Comment:
>  If it has been more than six hours since someone has generated the report
>  then a request will cause the report to be generated, correct?

I don't think so ... as I understand the consistency cache this is just the time that a regular check will be made to the fallback cubby cache: every six hours it will check if there is a difference between the in-memory consistency cache and the cubby cache.

We may need to have a chat to walk through this to make sure my logic is correct and matches reality, but I believe the way things are coded in this change mean that the only time the report is generated is:

1. when a job runs, or 
1. when an HTTP request is made and there is a cache miss